### PR TITLE
Centralized logging and more reproducibility

### LINKED ISSUE #30: More reproducibility
Say that I run:
```
bbpipe test.yml
```

which has stages `stage1` and `stage2`, and config file `config.yml`. Currently, it is not obvious to me that the information in `test.yml` and `config.yml` is fully saved for future reference. Also, the outputs from the two stages end up in the same directory by default, which makes it difficult to e.g. sync just some of the expensive stages to a different compute system. Here's a proposal for a slight restructuring.

`output_dir` will be the root directory. Pipeline products will be written to sub-directories in `output_dir` with the same name as the stages. `log_dir` does not have to be specified. Instead, it is always written to a file `$output_dir/run_{TIME}/log.txt`, where TIME is some identifier for the time bbpipe was run. Similarly, test.yml and config.yml are copied into `$output_dir/run_{TIME}/`. This will save more info about the submission. Let me know what you think and I can submit a PR.

### PR DESCRIPTION
**Pull request by msyriac**
This fixes part of #30 . Moves makedir commands from pipeline.py to main.py:run(). Removes log_dir and log_file options from configs. Make a run_{TIME} directory in output_dir, where TIME is current Unix time in milliseconds. Copies the pipeline yaml and stages config yaml in there. Makes run_{TIME} the directory where all logs go, including the pipeline log.

It now also saves stage outputs in their own sub-directories.

The 
```
bbpipe test/test.yml 
```
test runs fine after this.

### CODE REVIEW

**Comment by damonge** [comment]
Is this done? Let me know when you want me to take a look.

**Comment by msyriac** [comment]
Just adding one more feature (removing SCRATCH). Will switch out from draft soon.

**Comment by msyriac** [comment]
Never mind, I'll do the SCRATCH stuff in a separate PR. This ready for review @damonge .

File: bbpipe/pipeline.py [hunk 1]
```diff
@@ -100,16 +104,18 @@ def run(self, overall_inputs, output_dir, log_dir, resume, stages_config):
...
 
         for stage in stages:
             sec = self.stage_execution_config[stage.name]
-            app = stage.generate(self.dfk, sec.nprocess, sec.site, log_dir, mpi_command=self.mpi_command)
+            app = stage.generate(self.dfk, sec.nprocess, sec.site, log_dir,
+                                 mpi_command=self.mpi_command,
+                                 python_command=self.python_command)
             inputs = self.find_inputs(stage, data_elements)
+            stage_output_dir = f'{output_dir}/{stage.name}'
+            os.makedirs(stage_output_dir, exist_ok=True) # TODO: Maybe add a warning if it exists?
```
Comment (damonge) [inline bbpipe/pipeline.py:118]: Can we add this warning?

**Review by damonge — APPROVED** [review]
Looks good to me. I'd just add the warning before merging (ideally having merged the unit test branch first if you agree with the changes)

**Comment by msyriac** [comment]
Should be ready for merging now.

**Pull request merged**